### PR TITLE
TOOLS-2587: revendor MTC version 4.0.4 and fix qa-tests

### PR DESCRIPTION
**Pull request by huan-Mongo**

### CODE REVIEW

File: test/qa-tests/jstests/top/mongotop_sharded.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ var testName = 'mongotop_sharded';
     assert.strContains.soon(expectedError, getOutput, 'error message must appear at least once');
     var shellOutput = getOutput();
     jsTest.log('shell output: ' + shellOutput);
-    shellOutput.split('\n').forEach(function(line) {
+    shellOutput.split('\n').slice(1).forEach(function(line) {
       // check the displayed error message
       assert.neq(line.match(expectedError), null, 'unexpeced error message');
```
Comment (varsha804) [inline test/qa-tests/jstests/top/mongotop_sharded.js:16]: [nit] typo in "unexpected"

Comment (huan-Mongo) [reply test/qa-tests/jstests/top/mongotop_sharded.js:16]: ah, Thanks!

File: test/qa-tests/jstests/stat/stat_custom_headers.js [hunk 2]
```diff
@@ -20,36 +20,36 @@
...
-  }, "expected 5 rows in mongostat output");
-  assert.eq(statFields(rows[0]).join(), "host,conn,time",
-    "first row doesn't match 'host conn time'");
-  assert.eq(statFields(rows[1]).length, 5,
-    "there should be exactly five entries for a row of this stat output (time counts as three)");
+  }, "expected 6 rows in mongostat output");
+  assert.eq(statFields(rows[1]).join(), "host,conn,time",
+    "second row doesn't match 'host conn time'");
+  assert.eq(statFields(rows[2]).length, 5,
+    "third row should be exactly five entries for a row of this stat output (time counts as three)");
```
Comment (rychipman) [inline test/qa-tests/jstests/stat/stat_custom_headers.js:44]:
```suggestion
    "third row should have exactly five entries (time counts as three)");
```
this sentence didn't totally make sense as written

File: test/qa-tests/jstests/top/mongotop_sharded.js (hunk 1 above)
Comment (huan-Mongo) [reply test/qa-tests/jstests/top/mongotop_sharded.js:16]: I will change this before the merge. Don't wanna rerun the evergreen :-)

File: test/qa-tests/jstests/top/mongotop_json.js [hunk 3]
```diff
@@ -27,17 +27,17 @@ var testName = 'mongotop_json';
     }, 'invalid JSON 1');
 
     // ensure tool runs without error with --rowcount > 1
-    var rowcount = 5;
+    var rowcount = 8;
```
Comment (rychipman) [inline test/qa-tests/jstests/top/mongotop_json.js:30]: why'd this one jump to 8?

Comment (huan-Mongo) [reply test/qa-tests/jstests/top/mongotop_json.js:30]: Oh, this is a subjective number, currently, it should work with any value after the below change. I will change it back.

**Review by tfogo — COMMENTED** [review]
lgtm if tests are passing!

**Pull request merged**